--- proxyServer/live555ProxyServer.cpp	2013-01-25 18:47:20.000000000 +0000
+++ ../Github/live555/proxyServer/live555ProxyServer.cpp	2013-01-30 15:20:07.000000000 +0000
@@ -27,6 +27,7 @@
 int verbosityLevel = 0;
 Boolean streamRTPOverTCP = False;
 portNumBits tunnelOverHTTPPortNum = 0;
+portNumBits rtspServerPortNum = 554;
 char* username = NULL;
 char* password = NULL;
 
@@ -34,6 +35,7 @@
   *env << "Usage: " << progName
        << " [-v|-V]"
        << " [-t|-T <http-port>]"
+       << " [-p <rtsp-port>]"
        << " [-u <username> <password>]"
        << " <rtsp-url-1> ... <rtsp-url-n>\n";
   exit(1);
@@ -90,6 +92,22 @@
       break;
     }
 
+    case 'p': {
+      // set port
+      if (argc > 3 && argv[2][0] != '-') {
+	// The next argument is the RTSP server port number:                                                                       
+	if (sscanf(argv[2], "%hu", &rtspServerPortNum) == 1
+	    && rtspServerPortNum > 0) {
+	  ++argv; --argc;
+	  break;
+	}
+      }
+
+      // If we get here, the option was specified incorrectly:
+      usage();
+      break;
+    }
+
     case 'u': { // specify a username and password (to be used if the 'back end' (i.e., proxied) stream requires authentication)
       if (argc < 4) usage(); // there's no argv[3] (for the "password")
       username = argv[2];
@@ -131,12 +149,11 @@
   // access to the server.
 #endif
 
-  // Create the RTSP server.  Try first with the default port number (554),
+  // Create the RTSP server.  Try first with the default port number (554) or the one set by the caller,
   // and then with the alternative port number (8554):
   RTSPServer* rtspServer;
-  portNumBits rtspServerPortNum = 554;
   rtspServer = RTSPServer::createNew(*env, rtspServerPortNum, authDB);
-  if (rtspServer == NULL) {
+  if (rtspServer == NULL && rtspServerPortNum == 554) {
     rtspServerPortNum = 8554;
     rtspServer = RTSPServer::createNew(*env, rtspServerPortNum, authDB);
   }
